rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/ratelimiter/Limiter.java

```diff
@@ -0,0 +1,49 @@
+package com.lithium.dbi.rdbi.ratelimiter;
+
+import com.google.common.base.Throwables;
+
+import java.time.Duration;
+import java.util.OptionalLong;
+
+
+public interface Limiter {
+    /**
+     * Non-blocking method
+     *
+     * @return whether the permit was acquired or not.
+     */
+    boolean acquire();
+
+    /**
+     * Non blocking method
+     *
+     * @return If absent, the permit has been acquired. If present, indicates the time the client should wait before
+     * attempting to acquire permit again.
+     */
+    OptionalLong getWaitTimeForPermit();
+
+    /**
+     * Blocking method
+     * @param timeout
+     * @return true if acquired within the timeout, or false otherwise
+     */
+    default boolean acquirePatiently(Duration timeout) {
+        long timeWaited = 0L;
+        while (timeWaited < timeout.toMillis()) {
+            final OptionalLong waitTime = getWaitTimeForPermit();
+            if (!waitTime.isPresent()) {
+                return true;
+            } else {
+                final long timeToSleep = Math.max(100, waitTime.getAsLong());
```

Comment:
If this method is called with timeout = 5 seconds, and then getWaitTimeForPermit() returns 30 seconds, I think this implementation will incorrectly sleep for 30 seconds rather than 5 seconds.